pkg/cloud/aws/services/ec2/vpc.go

```diff
@@ -140,26 +152,36 @@ func (s *Service) createVPC() (*v1alpha1.VPCSpec, error) {
 	}
 	s.scope.V(2).Info("Created new VPC with cidr", "vpc-id", *out.Vpc.VpcId, "cidr-block", *out.Vpc.CidrBlock)
 
+	// TODO: we should attempt to record the VPC ID as soon as possible by setting s.scope.VPC().ID
+	// however, the logic used for determining managed vs unmanaged VPCs relies on the tags and will
+	// need to be updated to accommodate for the recording of the VPC ID prior to the tagging.
+
 	wReq := &ec2.DescribeVpcsInput{VpcIds: []*string{out.Vpc.VpcId}}
 	if err := s.scope.EC2.WaitUntilVpcAvailable(wReq); err != nil {
 		return nil, errors.Wrapf(err, "failed to wait for vpc %q", *out.Vpc.VpcId)
 	}
 
+	// Apply tags so that we know this is a managed VPC.
 	tagParams := s.getVPCTagParams(*out.Vpc.VpcId)
 	tagApply := &tags.ApplyParams{
 		EC2Client:   s.scope.EC2,
 		BuildParams: tagParams,
 	}
-
-	if err := tags.Apply(tagApply); err != nil {
-		return nil, err
+	tag := func() (bool, error) {
+		if err := tags.Apply(tagApply); err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+	if err := wait.WaitForWithRetryable(wait.NewBackoff(), tag, awserrors.VPCNotFound); err != nil {
+		record.Warnf(s.scope.Cluster, "FailedTagVPC", "Failed to tag managed VPC %q: %v", *out.Vpc.VpcId, err)
 	}
 
 	return &v1alpha1.VPCSpec{
 		ID:        *out.Vpc.VpcId,
 		CidrBlock: *out.Vpc.CidrBlock,
 		Tags:      v1alpha1.Build(tagParams),
-	}, nil
+	}, err
```

Comment:
Walk me through what happens if we return a valid VPCSpec with an error?

Reply:
- The calling function (reconcile) will treat it as an error and return an error to the Actuator ignoring the returned vpc information
- Since the returned vpc is not copied into the scope, the returned vpc data is basically dropped (as it is today).

This is essentially a noop from today just replacing the two independent returns (nil, error) and (vpc, nil) with a single return. If it would make it clearer that there are no side effects from this (which there shouldn't be if we are sticking to Go convention to ignore the return value if there is an error), then I'd be happy to revert back to two distinct returns.

Reply:
I think it's safer to have the two distinct returns